server/src/controllers/comments.controller.js

```diff
@@ -0,0 +1,148 @@
+import models from '../database/models';
+
+const { Articles, Comments } = models;
+/**
+   * create a new comment for an article
+   * @method createComment
+   * @param {object} req
+   * @param {object} res
+   * @returns {(function|object)} Function next() or JSON object
+   */
+export const createCommentOnArticle = async (req, res) => {
+  const { id: userId } = req.decoded;
+  const { id } = req.query;
+
+  const foundArticle = await Articles.findOne({ where: { id } });
+
+  if (!foundArticle) return res.status(404).json({ status: 'failure', error: 'article does not exist' });
+
+  const { id: articleId, type } = foundArticle;
+
+  if (type !== 'published') return res.status(400).json({ status: 'failure', error: 'you cannot comment on an unpublished article' });
```

Comment:
Nice check here. I think we should rename this article field from `type` to `status`